sdks/python/apache_beam/transforms/util_test.py

```diff
@@ -423,6 +425,70 @@ def test_reshuffle_streaming_global_window(self):
                 label='after reshuffle')
     pipeline.run()
 
+  @attr('ValidatesRunner')
+  def test_reshuffle_preserves_timestamps(self):
+    pipeline = TestPipeline()
+
+    # Create a PCollection and assign each element with a different timestamp.
+    before_reshuffle = (pipeline
+                        | "Four elements" >> beam.Create([
+                            {'name': 'foo', 'timestamp': MIN_TIMESTAMP},
+                            {'name': 'foo', 'timestamp': 0},
+                            {'name': 'bar', 'timestamp': 33},
+                            {'name': 'bar', 'timestamp': MAX_TIMESTAMP},
+                        ])
+                        | "With timestamp" >> beam.Map(
+                            lambda element: beam.window.TimestampedValue(
+                                element, element['timestamp'])))
+
+    # For each element in a PCollection, gets the current timestamp of the
+    # element and reassigns the timestamp to the element.
+    class AddTimestamp(beam.DoFn):
+      def process(self, element, timestamp=beam.DoFn.TimestampParam):
+        yield beam.window.TimestampedValue(element, timestamp)
+
+    # Reshuffle the PCollection above and assign the timestamp of an element to
+    # that element again.
+    after_reshuffle = (before_reshuffle
+                       | "Reshuffle" >> beam.Reshuffle()
```

Comment:
There's no need to name transforms (DRY) unless they happen again. In this case, the default name will already be Reshuffle. Likewise `Create` above is a sufficient name.

Reply:
I agree.
Done.